python_modules/dagster/dagster/components/core/load_defs.py

```diff
@@ -61,9 +70,51 @@ def get_project_root(defs_root: ModuleType) -> Path:
     raise FileNotFoundError("No project root with pyproject.toml or setup.py found")
 
 
+@public
+@preview(emit_runtime_warning=False)
+@suppress_dagster_warnings
+def load_from_defs_folder(*, project_root: Path) -> Definitions:
+    """Constructs a Definitions object, loading all Dagster defs in the project's
+    defs folder.
+
+    Args:
+        project_root (Path): The path to the dg project root.
+    """
+    root_config_path = discover_config_file(project_root)
+    toml_config = load_toml_as_dict(
+        check.not_none(
+            root_config_path,
+            additional_message=f"No config file found at project root {project_root}",
+        )
+    )
+
+    if root_config_path and root_config_path.stem == "dg":
+        project = toml_config.get("project", {})
+    else:
+        project = toml_config.get("tool", {}).get("dg", {}).get("project", {})
+
+    root_module_name = project.get("root_module")
+    defs_module_name = project.get("defs_module")
+    check.invariant(
+        defs_module_name or root_module_name,
+        f"Either defs_module or root_module must be set in the project config {root_config_path}",
+    )
```

Comment:
not blocking, but do we get better error messages in the underlying calls into dg core?